### PR TITLE
Use gtk3 instead of 2

### LINKED ISSUE #4: GTK3
I'd love to use this but my app already uses gtk3. Do you think I should make a pull request here or I would be better forking it?

### PR DESCRIPTION
**Pull request by meskio**
Closes #4

### CODE REVIEW

File: dlgs_linux.go [hunk 1]
```diff
@@ -4,41 +4,53 @@ import (
...
-
 func closeDialog(dlg *gtk.Dialog) {
 	dlg.Destroy()
-	/* The Destroy call itself isn't enough to remove the dialog from the screen; apparently
-	** that happens once the GTK main loop processes some further events. But if we're
-	** in a non-GTK app the main loop isn't running, so we empty the event queue before
-	** returning from the dialog functions.
-	** Not sure how this interacts with an actual GTK app... */
-	for gtk.EventsPending() {
-		gtk.MainIteration()
```
Comment (sqweek) [inline dlgs_linux.go]: Hm, can you elaborate on why this is no longer needed? Is it something to do with the switch to `glib.IdleAdd`? If that is doing what I think it is doing (submitting a function to be run as part of the main loop), does that break functionality for non-gtk apps?

I tried to test myself but I'm getting build errors upstream and don't have time to investigate myself, eg.:

```
# github.com/gotk3/gotk3/glib
../../gotk3/gotk3/glib/glib.go:260: cannot use idleSrc (type *_Ctype_GSource) as type *_Ctype_struct__GSource in function argument
../../gotk3/gotk3/glib/glib.go:283: cannot use timeoutSrc (type *_Ctype_GSource) as type *_Ctype_struct__GSource in function argument
../../gotk3/gotk3/glib/glib.go:294: cannot use src (type *_Ctype_struct__GSource) as type *_Ctype_GSource in function argument
```

Comment (meskio) [reply dlgs_linux.go]: I think they are related, yes. I remember we were having problems as we have other gtk parts in our app, and switched to send the dialogs into the main loop.

I see this approach doesn't work properly in programs that doesn't have a gtk main loop already running. I need to investigate more how to support both our needs and the non-gtk app needs.

**Comment by meskio** [comment]
Let's remove the sending of dialogs into the main loop, and go step by step.

I think now is just the migration to gtk3. And I'll explore the other improvements more before pushing them into a pull-req. Sorry for the noise.

**Comment by sqweek** [comment]
No apology necessary, thanks for doing the work! I've filed #25 to follow up on the gtk interaction.

**Pull request closed without merging**